lightning/src/ln/peers/handler.rs

```diff
@@ -30,9 +29,58 @@ use std::ops::Deref;
 use bitcoin::hashes::sha256::Hash as Sha256;
 use bitcoin::hashes::sha256::HashEngine as Sha256Engine;
 use bitcoin::hashes::{HashEngine, Hash};
-use ln::peers::handshake::PeerHandshake;
-use ln::peers::conduit::Conduit;
-use ln::peers::handshake::acts::Act;
+use ln::peers::outbound_queue::OutboundQueue;
+use ln::peers::transport::{PayloadQueuer, Transport};
+use bitcoin::hashes::core::iter::Filter;
+use std::collections::hash_map::IterMut;
+
+const MSG_BUFF_SIZE: usize = 10;
+
+/// Interface PeerManager uses to interact with the Transport object
+pub(super) trait ITransport: MessageQueuer {
+	/// Instantiate the new outbound Transport
+	fn new_outbound(initiator_static_private_key: &SecretKey, responder_static_public_key: &PublicKey, initiator_ephemeral_private_key: &SecretKey) -> Self;
```

Comment:
I think the default BOLT8  API requirement is still leaking in this abstract interface. The transport used may not be encrypted at all and use raw Internet address as node id.

Masking with an opaque `TransportState` object could be a solution but I don't advocate doing it now. Let's defer this in the future when someone wants actually to implement something different from BOLT8.

Reply:
I think that is prudent. I'll put an item in the backlog just to keep it around so we can review it over time. I'm a big fan of not over-engineering too much upfront if it is easy to fixup later when we need it.